images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,339 @@
+       returned: success
+       type: complex
+     kind:
+       description: Always 'Event'.
+       returned: success
+       type: str
+     lastTimestamp:
+       description: Timestamp of last occurance of Event
+       returned: success
+       type: timestamp
+     message:
+       description: Status for operation
+       returned: success
+       type: str
+     metadata:
+       description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+       returned: success
+       type: complex
+     reason:
+       description: Reason for the transition into the objects current status
+       returned: success
+       type: str
+     reportingComponent:
+       description: Component responsible for event
+       returned: success
+       type: str
+     source:
+       description: Component for reporting this Event
+       returned: success
+       type: string
+     type:
+       description: Type of Event. Either Normal or Warning
+       returned: success
+       type: string
+
+"""
+
+import copy
+import traceback
+import datetime
+import kubernetes.config.dateutil
+import openshift
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+# user args
+EVENT_ARG_SPEC = {
+    "state": {"default": "present", "choices": ["present", "absent"]},
+    "name": {"required": True},
+    "namespace": {"required": True},
+    "merge_type": {"type": "list", "choices": ["json", "merge", "strategic-merge"]},
+    "message": {"type": "str", "required": True},
+    "reason": {"type": "str", "required": True},
+    "reportingComponent": {"type": "str", "required": True},
+    "type": {"choices": ["Normal", "Warning"]},
+    "source": {"type": "str", "component": {"type": "str", "required": True}},
+    "involvedObject": {
+        "type": "dict",
+        "apiVersion": {"type": "str", "required": True},
+        "kind": {"type": "str", "required": True},
+        "name": {"type": "str", "required": True},
+        "namespace": {"type": "str", "required": True},
+    },
+}
+
+class KubernetesEvent(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubernetesEvent, self).__init__(*args, k8s_kind="Event", **kwargs)
+
+    @property
+    def argspec(self):
+        """ argspec property builder """
+        argumentSpec = copy.deepcopy(AUTH_ARG_SPEC)
+        argumentSpec.update(EVENT_ARG_SPEC)
+        return argumentSpec
+
+    def execute_module(self):
+        """ Module execution """
+        #auth
+        self.client = self.get_api_client()
+
+        message = self.params.get("message")
+        reason = self.params.get("reason")
+        reporting_component = self.params.get("reportingComponent")
+        event_type = self.params.get("type")
+        source = self.params.get("source")
```

Comment:
@chancez I feel like we should define these in `__init__` - what do you think?

Reply:
This could also be relegated to a follow-up PR

Reply:
Follow up

Reply:
What's here is fine